### ISSUE TITLE
Unknown plugin "transform-builtin-classes"

### ISSUE THREAD

**Issue by Sharcoux** [versatica/mediasoup-client#30]
On `mediasoup-client 2.0.17`

With `react-native 0.55.3`, I get this error when running the app on android:

```
error: bundling failed: ReferenceError: Unknown plugin "transform-builtin-classes" specified in "xxx/node_modules/mediasoup-client/.babelrc" at 0, attempted to resolve relative to "xxx/node_modules/mediasoup-client"
    at xxx/node_modules/babel-core/lib/transformation/file/options/option-manager.js:180:17
    at Array.map (<anonymous>)
    at Function.normalisePlugins (xxx/node_modules/babel-core/lib/transformation/file/options/option-manager.js:158:20)
    at OptionManager.mergeOptions (xxx/node_modules/babel-core/lib/transformation/file/options/option-manager.js:234:36)
    at OptionManager.init (xxx/node_modules/babel-core/lib/transformation/file/options/option-manager.js:368:12)
    at File.initOptions (xxx/yumii-mobile/node_modules/babel-core/lib/transformation/file/index.js:212:65)
    at new File (xxx/yumii-mobile/node_modules/babel-core/lib/transformation/file/index.js:135:24)
    at Pipeline.transform (xxx/yumii-mobile/node_modules/babel-core/lib/transformation/pipeline.js:46:16)
    at Object.transform (xxx/yumii-mobile/node_modules/metro/src/transformer.js:135:5)
```

However, I didn't encounter that error when using `react-native 0.50.4`

I'm quite unsure about what could cause that...

**Comment by Sharcoux** [comment]
You're right, adding `babel-plugin-transform-builtin-classes` to the devDependencies fixed the issue... But I don't get it, the plugin is in the devDependencies of mediasoup-client. Why is it needed in the client project? Moreover, why do I get this error only now that I upgraded react-native?

**Comment by ibc** [comment]
> But I don't get it, the plugin is in the devDependencies of mediasoup-client

This is how NPM works. `devDependencies` are not installed by default.

> Why is it needed in the client project?

Because an error in Babel. `mediasoup-client` includes a `.babelrc` file that fixes an error in Babel when it's bundled into a single `dist/xxxx` file.

> Moreover, why do I get this error only now that I upgraded react-native?

No idea, that's not mediasoup-client business.

**Comment by ibc** [comment]
Fixed in `mediasoup-client` 2.0.18
